src/dmd/cppmangle.d

```diff
@@ -138,6 +151,40 @@ private final class CppMangleVisitor : Visitor
         return -1;
     }
 
+    /**
+     * Overload which accepts a Namespace
+     *
+     * It is very common for large C++ projects to have multiple files sharing
+     * the same `namespace`. If any D project adopts the same approach
+     * (e.g. separating data structures from functions), it will lead to two
+     * `Nspace` objects being instantiated, with different addresses.
+     * At the same time, we cannot compare just any Dsymbol via identifier,
+     * because it messes with templates.
+     *
+     * See_Also:
+     *  https://issues.dlang.org/show_bug.cgi?id=18922
+     *
+     * Params:
+     *   ns = C++ namespace to do substitution for
+     *
+     * Returns:
+     *  Index of the entry, if found, or `-1` otherwise
+     */
+    int find(Nspace ns)
```

Comment:
It seems that this method is used solely inside nspace.d. If that is the case, make it private. Also, is there a possibility that the parameter can be made `const`? It doesn't seem to be modified.

Reply:
The whole class is `private final`.

Also for the parameter to be `const`, `const` needs to be added to `Nspace.equals`.
I think it should also be added to all base class (in D it wouldn't be a problem, but I'm not sure how that would play with the C++ header).